index.js

```diff
@@ -10,9 +10,14 @@ async function handleRequest(event) {
 	const request = event.request;
 	const url = new URL(request.url);
 
-	// Construct the cache key from the cache URL
-	const cacheKey = new Request(url.toString(), request);
-	const cache = caches.default;
+	// Construct the cache key from the cache URL, omitting sensitive headers
+	const cacheKey = new Request(url.toString(), {
+		method: request.method,
+		// Do not include Authorization or other sensitive headers
+	});
```

Comment:
```suggestion
	// Construct the cache key from the cache URL, omitting headers, which are variable (referrer includes the site URL)
	const cacheKey = new Request(url.toString());
```

The second parameter is optional. 

And most/all requests will be GET anyway? We would be better ignoring other at the top of the function than caching for unexpected requests.